### PR TITLE
Highlight scripts used by current scene

### LINKED ISSUE #10750: Highlight scripts used by current scene
### Describe the project you are working on

I was just tinkering with the engine and accidentally found a 9 year old feature which was later removed...

### Describe the problem or limitation you are having in your project

There is a fairly common confusion as to how script editor relates to opened scene. Usually in form of "add scripts to top tabs instead of it's own list" or something.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Highlight the scripts that are used by currently opened scene. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

highlighting
This feature is actually still fully implemented and functional (added in [e9bbb97](https://github.com/godotengine/godot/commit/e9bbb97acccc08ae03fde41e4cc6d2dc6722021a)), it's just overridden by 1 line of code (in [081a236](https://github.com/godotengine/godot/commit/081a236c6739e7f6d45731ec0c002d2269f2367b)). Just need to tweak the color, originally it was really attention grabbing and also covered the selected/hover color.
PS. custom icons in the screenshot above are unrelated here, I just grabbed it from my current build where I accidentally summoned the feature.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It's already existing feature and also one that cannot be implemented by addon. Either reimplement it or remove unused code if the feature is not wanted.

### PR DESCRIPTION
**Pull request by aXu-AP**
Closes godotengine/godot-proposals#10750
highlighting2

### CODE REVIEW

**Comment by Mickeon** [comment]
I see overwhelming positivity behind this PR (not the proposal though). Perhaps it's worth continuing. 

Although, we're still struggling to find a good, non-hardcoded theme colour for it.
Perhaps it could be the highlight colour, but heavily mixed with the default background colour.

**Comment by aXu-AP** [comment]
Transparency isn't only for dimmer look, it also is needed for regular highlighting to work (it seems that item background is drawn on top of highlight box).
Here's `highlight` color with transparency down to 20%, various themes:
kuva
`ai_component` is the script selected, `mouse` is hovering over player. `enemy` and `health_component` are in the current scene.
Looks not bad in the default theme. In light theme, the color is quite close to mouse hover color. In black theme highlight is quite dim.
I think it's acceptable? I'm not too worried about being close to hover color. In motion it looks better.

**Comment by Calinou** [comment]
>  Current PR uses transparency for a dimmer look, but I don't know if that'll play nice on light themes.

You can use a transparent gray (`Color(0.5, 0.5, 0.5, 0.1)`) which will work well on any background color.

**Comment by Mickeon** [comment]
I like the chosen look in the above screenshots, actually. I am concerned that the background on the fully black theme being completely invisible, though.

Having it be subtle but noticeable is nice. As you said, it looks better in motion. It pops out when changing scenes around but it's not distracting.

A friend has also expressed concern since this is restoring a feature users have gotten used to without. I believe most would see this as an upgrade, but it may warrant an Editor Setting, we'll see.

**Comment by aXu-AP** [comment]
Here's with transparent gray as suggested by @Calinou:
kuva
I think it's tad more readable.
Editor setting sounds reasonable, as we have it with script temperature also. Some do like more vanilla list.

**Comment by aXu-AP** [comment]
Changed the color to `Color(.5, .5, .5, .1)` and added editor setting `text_editor/script_list/highlight_scene_scripts` now.

File: doc/classes/EditorSettings.xml [hunk 1]
```diff
@@ -1307,6 +1307,9 @@
 		<member name="text_editor/script_list/group_help_pages" type="bool" setter="" getter="">
 			If [code]true[/code], class reference pages are grouped together at the bottom of the Script Editor's script list.
 		</member>
+		<member name="text_editor/script_list/highlight_scene_scripts" type="bool" setter="" getter="">
+			If [code]true[/code], the scripts that are used by currently opened scene are highlighted.
```
Comment (Mickeon) [inline doc/classes/EditorSettings.xml:1311]:
```suggestion
			If [code]true[/code], the scripts that are used by currently opened scene are highlighted in the Script Editor's script list.
```

Comment (AThousandShips) [reply doc/classes/EditorSettings.xml:1311]: Missing a "the" before "current scene"

**Review by Mickeon — APPROVED** [review]
I'd say yeah as is. It's either this, or the feature is stripped out of the codebase because it currently is not working.

I have a gut feeling that the highlight will miss _some_ scripts in the current scene, but we'll see. There's a huge desire to rework the script list someday, but that day is not as of writing this.

**Review by Calinou — COMMENTED** [review]
Tested locally, it works as expected. Built-in scripts work, as well as scripts from instanced subscenes (they will be highlighted as being part of the current scene, which is technically valid).

The highlighting is a bit too subtle for me to easily notice it though. I suggest increasing the opacity to `0.125`.

**Comment by aXu-AP** [comment]
Changed it to 0.125, I think it's about right. Comparison with varying degrees of opacity:
kuva

**Review by Calinou — APPROVED** [review]
Looks great!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**